packages/calcite-components/src/components/input-number/input-number.tsx

```diff
@@ -1095,8 +1095,8 @@ export class InputNumber
         autofocus={this.el.autofocus ? true : null}
         defaultValue={this.defaultValue}
         disabled={this.disabled ? true : null}
-        enterKeyHint={this.el.enterKeyHint || this.el.getAttribute("enterkeyhint")}
-        inputMode={this.el.inputMode || this.el.getAttribute("inputmode") || "decimal"}
+        enterKeyHint={this.el.enterKeyHint}
```

Comment:
`enterkeyhint` attribute reflects to `enterKeyHint` propertie.
Verified in Chrome, Firefox and Safari.

Same for others

Reply:
EDIT: that is true for `<input>` elements.
it is not true for custom elements.
reverting this change